host/sgx/loadelf.c

```diff
@@ -834,6 +1019,155 @@ static oe_result_t _sgx_update_enclave_properties(
     return result;
 }
 
+static oe_result_t _find_module_image(
+    const char* enclave_path,
+    oe_enclave_image_t* image,
+    char** module_path)
+{
+    oe_result_t result = OE_OK;
+    elf64_dyn_t* section_data = NULL;
+    size_t section_size = 0;
+    uint64_t number_of_entries;
+    elf64_addr_t strtab_offset = 0;
+    elf64_xword_t needed_offset = 0;
+    char* module_name = NULL;
+    char* path_name = NULL;
+    size_t path_size = 0;
+    const char* p = NULL;
+
+    assert(enclave_path);
+    assert(image);
+
+    if (module_path)
+        *module_path = NULL;
+
+    if (elf64_find_section(
+            &image->elf.elf,
+            ".dynamic",
+            (uint8_t**)&section_data,
+            &section_size) != 0)
+        goto done;
+
+    if (!section_data || !section_size)
+        OE_RAISE(OE_UNEXPECTED);
+
+    number_of_entries = section_size / sizeof(elf64_dyn_t);
+    for (uint64_t i = 0; i < number_of_entries; i++)
+    {
+        /* Explicitly prevent the use of DT_RPATH and DT_RUNPATH that affects
+         * the enclave measurement */
+        if (section_data[i].d_tag == DT_RPATH ||
+            section_data[i].d_tag == DT_RUNPATH)
+            OE_RAISE_MSG(
+                OE_UNEXPECTED,
+                "RPATH or RUNPATH should not be used in the enclave binary",
```

Comment:
We can specify the compiler/linker/cmake flags to avoid RPATH/RUNPATH in the error message